rholang/src/main/scala/coop/rchain/rholang/crypto/Blake2b512Block.scala

```diff
@@ -0,0 +1,152 @@
+package coop.rchain.rholang.crypto
+
+import coop.rchain.rspace.Serialize
+import org.bouncycastle.util.Pack
+
+/**
+Block oriented Blake2b512 class.
+
+Only supports hashes of data a positive number of whole blocks.
+  */
```

Comment:
You've based this off `Blake2bDigest.java` in bouncycastle, no?

Might be good to link to that code, especially to help ascertain the provenance of your constants like `SIGMA` and `IV`.